hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -231,7 +231,13 @@ private static Option<Schema.Field> findNestedField(Schema schema, String[] fiel
     if (!nestedPart.isPresent()) {
       return Option.empty();
     }
-    return nestedPart;
+    boolean isUnion = false;
```

Comment:
Could you explain this change?  This affects the logic of all callers, not just schema evolution.

Reply:
Before, this just found the field and returned it. Now it returns the entire path to the field.

so if the record was 

{
a: string
b: {
    x: long
    y:int 
   }
}

before, the method would just return "y:int".

Now, the method will return "b:{y:int}".

this is so when we call appendFieldsToSchemaDedupNested we can dedup the field if it already exists.


For example, before if we tried to append "y:int" to the schema we would get 


{
a: string
b: {
    x: long
    y:int 
   }
y:int
}

now, that we append "b:{y:int}" we will get 

{
a: string
b: {
    x: long
    y:int 
   }
}